src/lib/sidenav/sidenav.ts

```diff
@@ -38,6 +38,7 @@ export class MdDuplicatedSidenavError extends MdError {
   template: '<ng-content></ng-content>',
   host: {
     '(transitionend)': '_onTransitionEnd($event)',
+    '(keydown.escape)': 'handleEscapeKey()',
```

Comment:
unfortunately we can't use this fancy syntax, since not all of the browsers we target support the `key` property of `KeyboardEvent`. We have a file of keycodes here: https://github.com/angular/material2/blob/master/src/lib/core/keyboard/keycodes.ts